tests/shared/drive-constraints.spec.js

```diff
@@ -771,4 +772,233 @@ describe('Shared: DriveConstraints', function() {
 
   });
 
+  describe('.getDriveImageCompatibilityStatuses', function() {
+
+    beforeEach(function() {
+      if (process.platform === 'win32') {
+        this.mountpoint = 'E:';
+        this.separator = '\\';
+      } else {
+        this.mountpoint = '/mnt/foo';
+        this.separator = '/';
+      }
+
+      this.drive = {
+        device: '/dev/disk2',
+        name: 'My Drive',
+        protected: false,
+        system: false,
+        mountpoints: [
+          {
+            path: this.mountpoint
+          }
+        ],
+        size: 4000000000
+      };
+
+      this.image = {
+        path: path.join(__dirname, 'rpi.img'),
+        size: {
+          original: this.drive.size - 1,
+          final: {
+            estimation: false
+          }
+        }
+      };
+    });
+
+    const expectStatusTypesAndMessagesToBe = (resultList, expectedTuples) => {
+
+      // Sort so that order doesn't matter
+      const expectedTuplesSorted = _.sortBy(_.map(expectedTuples, (tuple) => {
+        return {
+          type: constraints.COMPATIBILITY_STATUS_TYPES[tuple[0]],
+          message: constraints.COMPATIBILITY_STATUS_MESSAGES[tuple[1]]
+        };
+      }), [ 'message' ]);
+      const resultTuplesSorted = _.sortBy(resultList, [ 'message' ]);
+
+      m.chai.expect(resultTuplesSorted).to.deep.equal(expectedTuplesSorted);
+    };
+
+    describe('given there are no errors or warnings', () => {
+
+      it('should return an empty list', function() {
+        const result = constraints.getDriveImageCompatibilityStatuses(this.drive, {
+          path: '/mnt/disk2/rpi.img',
+          size: 1000000000
+        });
+
+        m.chai.expect(result).to.deep.equal([]);
+      });
+
+    });
+
+    describe('given the drive contains the image', () => {
+
+      it('should return the contains-image error', function() {
+        this.image.path = `${this.mountpoint}${this.separator}rpi.img`;
```

Comment:
Probably better to do something similar to what https://github.com/resin-io/etcher/blob/master/tests/shared/drive-constraints.spec.js#L134 does, rather than manually building paths by concatenating strings.